vyper/semantics/types/function.py

```diff
@@ -22,7 +22,53 @@
 from vyper.semantics.types.shortcuts import UINT256_T
 from vyper.semantics.types.subscriptable import TupleT
 from vyper.semantics.types.utils import type_from_abi, type_from_annotation
-from vyper.utils import OrderedSet, keccak256
+from vyper.utils import MemoryPositions, OrderedSet, keccak256, mkalphanum
+
+FunctionSignatures = Dict[str, "ContractFunctionT"]
+
+
+@dataclass
+class FunctionArg:
+    name: str
+    typ: VyperType
+    ast_source: Optional[vy_ast.VyperNode] = None
+    default_value: Optional[vy_ast.VyperNode] = None
+
+
+@dataclass
+class FrameInfo:
+    frame_start: int
+    frame_size: int
+    frame_vars: Dict[str, Tuple[int, VyperType]]
+
+    @property
+    def mem_used(self):
+        return self.frame_size + MemoryPositions.RESERVED_MEMORY
+
+
+@dataclass
+class FunctionIRInfo:
+    identifier: str
+    frame_info: Optional[FrameInfo] = None
+
+    @property
+    def exit_sequence_label(self) -> str:
+        return self.identifier + "_cleanup"
+
+
+@dataclass
```

Comment:
separating these into different classes seems a little bit pedantic but it might be ok